types/app.d.ts

```diff
@@ -123,6 +122,9 @@ declare namespace Realm {
      * A MongoDB Realm App.
      */
     class App<FunctionsFactoryType extends object = DefaultFunctionsFactory> {
+
+        constructor(AppConstructor);
```

Comment:
Shouldn't this be taking an `AppConfiguration` and not the `AppConstructor` type here? Did you test this manually?